base_layer/wallet_ffi/src/lib.rs

```diff
@@ -1859,12 +1897,44 @@ pub unsafe extern "C" fn wallet_test_mined(
         ptr::swap(error_out, &mut error as *mut c_int);
         return false;
     }
-    if tx.is_null() {
-        error = LibWalletError::from(InterfaceError::NullError("tx".to_string())).code;
+
+    match finalize_received_transaction(&mut *wallet, tx_id) {
+        Ok(_) => true,
+        Err(e) => {
+            error = LibWalletError::from(e).code;
+            ptr::swap(error_out, &mut error as *mut c_int);
+            false
+        },
+    }
+}
+
+/// This function will simulate the process when a completed transaction is broadcast to
+/// the base layer mempool. The function will update the status of the completed transaction
+///
+/// ## Arguments
+/// `wallet` - The TariWallet pointer
+/// `tx_id` - The TxId of the transaction to operate on
+/// `error_out` - Pointer to an int which will be modified to an error code should one occur, may not be null. Functions
+/// as an out parameter.
+///
+/// ## Returns
+/// `bool` - Returns if successful or not
+#[no_mangle]
+pub unsafe extern "C" fn wallet_test_broadcast_transaction(
+    wallet: *mut TariWallet,
+    tx_id: c_ulonglong,
```

Comment:
Consider a pointer a transaction struct instead of c_ulonglong for the same reasons as above

Reply:
done